File: arrays/matrix_diagonal_sum.rb
# @param {Integer[][]} mat
# @return {Integer}
def diagonal_sum(mat)
    n = mat.size
    sum = 0

    (0...n).each do |i|
        sum += mat[i][i]
        sum += mat[i][-(1 + i)] 
    end

    center = n / 2
    return n.odd? ? sum -= mat[center][center] : sum
end

File: arrays/matrix_diagonal_sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        n = len(mat)
        total = 0
    
        for i in range(n):
            total += mat[i][i]
            total += mat[i][n - 1 - i]  
    
        if n % 2 == 1:
            center = n // 2
            total -= mat[center][center]
    
        return total